### ISSUE TITLE
ARK creation upon ingest

### ISSUE THREAD

**Issue by jcoyne** [curationexperts/alexandria-v2#34]
From Ian:
> We are using ARK identifiers for all our digital objects in ADRL and we want to do so for images as well.
>
> We’re using the EZID web service from CDL to create the objects.
>
> We can discuss this on our next call but I thought I should let you know about it right away and I apologize that we didn’t bring this up earlier.
>
> Some requirements:
>     A new ARK will be minted for each new digital object as it is ingested into the repository.
>     Location, Who, What & When metadata injected into EZID’s system.
>     ARK used as part of Digital Object’s Permalink

> Links:
>     The EZID system:  http://ezid.cdlib.org

>     A sample ETD ARK: ark:/48907/f3zc80sm
>     Permalink with this ARK:   http://alexandria.ucsb.edu/lib/ark:/48907/f3zc80sm
>     the same ARK at EZID:   http://ezid.cdlib.org/id/ark:/48907/f3zc80sm

**Comment by jcoyne** [comment]
Currently there are two show paths:
http://alexandria.ucsb.edu/catalog/adrl:f3zc80sm
and
http://alexandria.ucsb.edu/lib/ark:/48907/f3zc80sm
Maybe we can consolidate them

**Comment by jcoyne** [comment]
http://ezid.cdlib.org/doc/apidoc.html

https://github.com/duke-libraries/ezid-client
https://github.com/projecthydra-labs/hydra-remote_identifier

**Comment by jcoyne** [comment]
The record show page should be the ARK url: e.g. http://alexandria.ucsb.edu/lib/ark:/48907/f3zc80sm

The minted ark should be the fedora id.

**Comment by mark-dce** [comment]
We cannot consolidate the show paths - one is for the full object, one is for metadata (and possibly thumbnail harvesting) per Chrissy from Jan 20th -

Good morning,

For our ETDs, each ARK was added to the corresponding MODS record in two ways: 

1) As a non-resolvable persistent identifier for the resource:

<mods:identifier type="ark">ark:/48907/f3bg2kw5</mods:identifier>

2) As two resolvable permalink URLs. The first URL element records the link to the object within the context of ADRL (with associated metadata, navigation, etc.); the access attribute indicates the link that is most appropriate to display for end users (for OAI harvesters). This link is displayed to the user on the object page and is also the one intended for citations, bookmarks, etc.

The second URL element is supposed to indicate a link to a thumbnail (or snippet of text) and is provided primarily for OAI harvesters (especially CDL). This link is not displayed to the user on the object page.

<mods:location>
	<mods:url usage="primary" access="object in context">http://alexandria.ucsb.edu/lib/ark:/48907/f3bg2kw5</mods:url>
	<mods:url access="preview">http://alexandria.ucsb.edu/lib/ark:/48907/f3bg2kw5/thumbnail</mods:url>
</mods:location>

Please let me know if you have any questions.

Thanks,
Chrissy

**Comment by jcoyne** [comment]
@mark-dce I think currently there are more than two show paths. I can see we need a different path for the "show" and the "thumbnail", but I don't think we need two paths for the "show" page.

**Comment by mark-dce** [comment]
@jcoyne Sorry, I didn't look carefully enough at your example routes.  You're right, there should only be one route and for now I would like it to be /catalog/ark:/48907/id.  

I have created three tickets to clarify future requirements for 'lib' vs. 'catalog' in the path #80 (admin backlog); to clarify the '/mods:url' suffix requirements #81 (discovery backlog); and confirm whether one or two  routes to the same resource are required #82 (discovery backlog) .

I'm confused, however, was this story originally supposed to be about getting credentials for and connecting to the EZID server, minting ARKs, or routing?

**Comment by jcoyne** [comment]
@mark-dce the confusion stems from the fact that you need to register a URL when you mint the ark, so the question was, which one to register.

**Comment by mark-dce** [comment]
@jcoyne it doesn't look like we currently have a route set up for /lib, so if we just register the default route for the resource (/catalog/blah) does that work for now?  Or do we need to resolve the route naming question #82 before we can implement the registration code?

**Comment by jcoyne** [comment]
We can do something now, but the URL may be wrong. This ticket is really blocked by needing the credentials for ezid.

**Comment by acozine** [comment]
Dev/Test connection info for EZID:

development:
  ezid_url:            https://ezid.cdlib.org/
  ezid_user:           apitest
  ezid_password:       apitest
  ezid_post_uri:       /shoulder/ark:/99999/fk4

  institute_id:        99999

test:
  ezid_url:            https://ezid.cdlib.org/
  ezid_user:           apitest
  ezid_password:       apitest
  ezid_post_uri:       /shoulder/ark:/99999/fk4

  institute_id:        99999

**Comment by mark-dce** [comment]
@jcoyne - it might be worth taking a look at PSU's ez-id in addition to Dukes and Hydra-labs - https://github.com/psu-stewardship/hydra-ezid

**Comment by acozine** [comment]
I've put a shared/config/ingest.yml file on the sandbox server with the development connection info for EZID - the section header in the yaml file says "production" because the machine is set to production mode - however, the credentials are for the development EZID connection

**Comment by mark-dce** [comment]
Superseded by #100
